sbt-metadoc/src/main/scala/metadoc.sbt/MetadocPlugin.scala

```diff
@@ -0,0 +1,101 @@
+package metadoc.sbt
+
+import sbt._
+import sbt.Keys._
+
+/**
+  * Generate a metadoc site.
+  *
+  * == Usage ==
+  *
+  * This plugin must be explicitly enabled. To enable it add the following line
+  * to your `.sbt` file:
+  * {{{
+  * enablePlugins(MetadocPlugin)
+  * }}}
+  *
+  * By default, semantic data is read from `metadocClasspath` which is
+  * automatically populated based on the various filter settings.
+  */
+object MetadocPlugin extends AutoPlugin {
+
+  object autoImport {
+    val Metadoc = config("metadoc")
+    val metadocScopeFilter =
+      settingKey[ScopeFilter]("Control sources to be included in metadoc.")
+    val metadocProjectFilter = settingKey[ScopeFilter.ProjectFilter](
+      "Control projects to be included in metadoc."
+    )
+    val metadocConfigurationFilter =
+      settingKey[ScopeFilter.ConfigurationFilter](
+        "Control configurations to be included in metadoc."
+      )
+    val metadocClasspath =
+      taskKey[Seq[Classpath]]("Class directories to be included in metadoc")
+    val metadoc = taskKey[File]("Generate GraphQL API code")
+  }
+  import autoImport._
+
+  override def requires = plugins.JvmPlugin
+
+  lazy val classpathTask = Def.taskDyn {
+    fullClasspath.all(metadocScopeFilter.value)
+  }
+
+  override def projectSettings: Seq[Setting[_]] = Seq(
+    ivyConfigurations += Metadoc,
+    libraryDependencies ++= List(
+      // Explicitly set the Scala version dependency so the resolution doesn't pick
```

Comment:
The alternative would be to publish an artifact of the CLI using `coursier bootstrap` (or sbt-assembly) which can be picked out using the scope and run without additional dependencies resolved by sbt.